src/CodeGen_RISCV.cpp

```diff
@@ -21,11 +21,28 @@ string CodeGen_RISCV::mcpu() const {
 }
 
 string CodeGen_RISCV::mattrs() const {
-    return "";
+    string arch_flags = "+m,+a,+f,+d,+c";
+    if (target.has_feature(Target::RVV)) {
+        arch_flags += ",+experimental-v";
+    }
+    return arch_flags;
+}
+
+string CodeGen_RISCV::mabi() const {
+    string abi;
+    if (target.bits == 32) {
+        abi = "lp32";
+    } else {
+        abi = "lp64";
+    }
+    if (!target.has_feature(Target::SoftFloatABI)) {
+        abi += "d";
+    }
+    return abi;
 }
 
 bool CodeGen_RISCV::use_soft_float_abi() const {
-    return false;
+    return target.has_feature(Target::SoftFloatABI);
```

Comment:
Interesting, I didn't realize that RISC-V offered soft-float -- does this work? (I wouldn't be surprised if we have code somewhere that assumes it applies only to arm32, which AFAIK was this only place this was enabled previously)

Reply:
I added it just in case someone want the generated library compatible with soft-float ABI.
It works fine as I tried `test/generator/example.generator` with `target=riscv-64-noos-rvv-soft_float_abi`.